src/services/lostAndFound.ts

```diff
@@ -0,0 +1,138 @@
+import http from './http';
+import userService from './user';
+
+/**
+ * Missing item report object, representing the model of a report for communication with the
+ * backend service.
+ */
+export type MissingItemReport = {
+  recordID?: number;
+  firstName?: string;
+  lastName?: string;
+  category: string;
+  colors: string[]; // Ensures colors is always an array
+  brand?: string;
+  description: string;
+  locationLost: string;
+  stolen: boolean;
+  stolenDescription?: string;
+  dateLost: string;
+  dateCreated: string;
+  phone?: string;
+  email?: string;
+  status: string;
+  submitterUsername: string;
+  submitterID?: null;
+  forGuest: boolean;
+  adminActions?: MissingAdminAction[];
+};
+
+/**
+ * Missing Admin Action object, representing the model of an action taken on a missing item
+ * report for communication with the backend.
+ */
+export type MissingAdminAction = {
+  ID?: number;
+  missingID: number;
+  action: string;
+  actionDate: string;
+  actionNote?: string;
+  username: string;
+  isPublic?: boolean;
+};
+
+/**
+ * Fetch an array containing the full list of all missing item reports.
+ * @returns MissingItemReport[] array of all missing item reports.
+ */
+const getMissingItemReports = (): Promise<MissingItemReport[]> =>
+  http.get<MissingItemReport[]>('lostandfound/missingitems');
+
+/**
+ * Fetch an array containing the full list of all missing item reports for
+ * the currently logged in user.
+ * @returns MissingItemReport[] array of all missing item reports.
+ */
+const getMissingItemReportUser = (): Promise<MissingItemReport[]> => {
+  const profile = userService.getProfileInfo();
+  return profile.then((profileValue) =>
+    http.get<MissingItemReport[]>(`lostandfound/missingitems?user=${profileValue?.AD_Username}`),
+  );
+};
```

Comment:
I think it would be better to make the username a parameter of this function, so that it can be re-used for getting reports for a given user other than the current one. Also, `userService.getProfileInfo` sends ~5 API requests and does a lot of work to build the current user's whole profile, so this route becomes much less efficient than it could be.

It should be almost trivial to call `useUser` in the components that call `getMissingItemReportUser` so that we can pass in `profile.AD_Username` as a parameter.

Finally, putting URL Search Params in an interpolated string is bad practice, because URL params have to be encoded carefully. 

```suggestion
const getMissingItemReportUser = (username: string): Promise<MissingItemReport[]> => 
    http.get<MissingItemReport[]>(`lostandfound/missingitems?${http.toQueryString({user: username})}`)
```

Reply:
Just resolved this in the most recent merge from senior-project-2.